architecture/evm/hooks.go

```diff
@@ -50,6 +50,10 @@ func HandleUpstreamPreForward(ctx context.Context, n common.Network, u common.Up
 
 	switch method {
 	case "eth_getLogs":
+		// If the request has a blockhash, we can do a simple forward to upstream without sending it to the eth_getLogs hooks
+		if requestHasBlockHash(r) {
+			return false, nil, nil
+		}
```

Comment:
better move this logic inside the upstreamPreForward_eth_getLogs itself... this hook orchestration code doesn't need any logic leaked into it